google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreImpl.java

```diff
@@ -384,7 +401,21 @@ public void onSuccess(List<WriteResult> writeResults) {
                   @Override
                   public void run() {
                     try {
-                      callbackResult.set(transactionCallback.updateCallback(transaction));
+                      ApiFuture<T> updateCallback = transactionCallback.updateCallback(transaction);
+                      ApiFutures.addCallback(
```

Comment:
If I am reading the documentation of `addCallback` correctly in only runs the `ApiFutureCallback` on the provided executor. We actually want to run `transactionCallback` on the user executor. Do you mind fixing that and/or convincing me that I am mistaken?

Reply:
You are right.